docs/source/testing/testing.md

```diff
@@ -3,26 +3,55 @@ title: Integration testing
 description: Utilities for testing Apollo Server
 ---
 
-Testing `apollo-server` can be done in many ways. One simple way is to use ApolloServer's `executeOperation` method to directly execute a GraphQL operation without going through a full HTTP operation.
-
-## `executeOperation`
-
-Integration testing a GraphQL server means testing many things. `apollo-server` has a request pipeline that can support many plugins that can affect the way an operation is executed. The `executeOperation` method provides a single hook to run operations through the request pipeline, enabling the most thorough tests possible without starting up an HTTP server.
+Integration testing a GraphQL server means testing the combination of many interlocking parts at once. One way to test your server setup is to use `ApolloServer`'s `executeOperation` method to directly execute a GraphQL operation without going through a full HTTP request.
+
+## Testing using `executeOperation`
+
+The `apollo-server` library has a request pipeline that can support many plugins which can each affect the way an operation is executed. The `executeOperation` method is available to run operations through the request pipeline, enabling the most thorough tests possible without starting up an HTTP server.
+
+The [`executeOperation` method](../api/apollo-server/#executeoperation) accepts the following arguments:
+* An object containing configuration options, which must include a `query` option that specifies the GraphQL operation to run.
+  * You can use `executeOperation` to execute both queries and mutations, but both use the `query` option.
+* An argument that is passed in to the `ApolloServer` instance's [`context` function](../data/resolvers/#the-context-argument).
+
+Below is a simplified example of setting up a test using the JavaScript testing library [Jest](https://jestjs.io/):
+```js:title=index.test.js
```

Comment:
I'd suggest that when possible (at least in the future) when adding new code examples we do it in TypeScript rather than JS.  We have a cool `<MultiCodeBlock>` thing that lets users convert to JS. (I'm hoping that as part of the AS4 work we can overhaul all our docs to be TS-first. https://github.com/apollographql/apollo-server/issues/6052)

Another nice thing we've been trying to do is to have larger examples also exist in the https://github.com/apollographql/docs-examples repo so you can link to a functioning code sandbox for them. @StephenBarlow can help with that (and advise on whether it's useful here or not)

Reply:
Thank you for the context about the `< MultiCodeBlock>`. I've seen those in a couple of the docs examples—so snazzy! I'm a TypeScript newbie, so I will focus on strengthening my TS skills so in the future I can start writing TS examples.